packages/pdf-viewer/Page.tsx

```diff
@@ -62,52 +65,51 @@ export default function Page(props: PageProps) {
 	const textRef = useRef<HTMLDivElement>(null);
 	const wrapperRef = useRef<HTMLDivElement>(null);
 	const isVisible = useIsVisible(wrapperRef, props.container);
+	const currentMarkupState = useRef(props.markupState);
 	useVisibleOnSelect({
 		isVisible,
 		isSelected: props.isSelected,
 		container: props.container,
 		wrapperRef,
 	} as VisibleOnSelect);
 
-	useEffect(() => {
+	const renderPage = useCallback(async () => {
 		const isCancelled = () => props.scaledSize.scale !== scaleRef.current;
-
-		const renderPage = async () => {
-			try {
-				if (canvasRef.current) {
-					canvasRef.current.style.height = '100%';
-					canvasRef.current.style.width = '100%';
-				}
-				const renderRequest: RenderRequest = {
-					pageNo: props.pageNo,
-					scaledSize: props.scaledSize,
-					getTextLayer: props.textSelectable,
-					isCancelled,
-				};
-				const { canvas, textLayerDiv } = await props.pdfDocument.renderPage(renderRequest);
-
-				wrapperRef.current.appendChild(canvas);
-				if (textLayerDiv) wrapperRef.current.appendChild(textLayerDiv);
-
-				if (canvasRef.current) canvasRef.current.remove();
-				if (textRef.current) textRef.current.remove();
-
-				canvasRef.current = canvas;
-				if (textLayerDiv) textRef.current = textLayerDiv;
-			} catch (error) {
-				if (isCancelled()) return;
-				error.message = `Error rendering page no. ${props.pageNo}: ${error.message}`;
-				setError(error);
-				throw error;
+		try {
+			if (canvasRef.current) {
+				canvasRef.current.style.height = '100%';
+				canvasRef.current.style.width = '100%';
 			}
-		};
+			const renderRequest: RenderRequest = {
+				pageNo: props.pageNo,
+				scaledSize: props.scaledSize,
+				getTextLayer: props.textSelectable,
+				isCancelled,
+			};
+			const { canvas, textLayerDiv } = await props.pdfDocument.renderPage(renderRequest);
+
+			wrapperRef.current.appendChild(canvas);
+			if (textLayerDiv) wrapperRef.current.appendChild(textLayerDiv);
+
+			if (canvasRef.current) canvasRef.current.remove();
+			if (textRef.current) textRef.current.remove();
+
+			canvasRef.current = canvas;
+			if (textLayerDiv) textRef.current = textLayerDiv;
+		} catch (error) {
+			if (isCancelled()) return;
+			error.message = `Error rendering page no. ${props.pageNo}: ${error.message}`;
+			setError(error);
+			throw error;
```

Comment:
Won't that crash the viewer? Because it seems you're already handling it with the setError call so maybe it doesn't need to be rethrown?